packages/core/src/fhirpath/utils.ts

```diff
@@ -159,6 +159,7 @@ export function getTypedPropertyValueWithSchema(
   // In the case of [x] choice-of-type, the type must be resolved to a single type.
   if (primitiveExtension) {
     if (Array.isArray(resultValue)) {
+      resultValue = resultValue.slice();
```

Comment:
Minor request to add a comment here

```
// safeAssign can turn a primitive `string` into a `String`
// slice() ensures that the result is always a primitive `string`
```

Reply:
The `slice()` doesn't ensure the result is a primitive string, it prevents the original input `resultValue` array from having its entries changed.